x/shield/keeper/purchase.go

```diff
@@ -156,10 +156,15 @@ func (k Keeper) IterateAllPurchases(ctx sdk.Context, callback func(purchase type
 
 // RemoveExpiredPurchasesAndDistributeFees removes expired purchases and distributes fees for current block.
 func (k Keeper) RemoveExpiredPurchasesAndDistributeFees(ctx sdk.Context) {
+	lastUpdateTime, found := k.GetLastUpdateTime(ctx)
+	if !found || lastUpdateTime.IsZero() {
+		k.SetLastUpdateTime(ctx, ctx.BlockTime())
```

Comment:
This should be set at the first purchase time.

Reply:
Fixed.